e2e-tests/src/test/treasury.rs

```diff
@@ -38,11 +61,16 @@ pub fn channeling_fee(config: Config) -> anyhow::Result<()> {
         treasury_balance_after, issuance_after
     );
 
-    assert!(
-        issuance_after <= issuance_before,
-        "Unexpectedly {} was minted",
-        issuance_after - issuance_before
-    );
+    if issuance_after > issuance_before {
+        let difference = issuance_after - issuance_before;
+        assert_eq!(
+            difference % possibly_treasury_gain_from_staking,
+            0,
```

Comment:
Uh? Shouldn't you just compare the two above, rather than taking modulo?

Reply:
Author (myself) reasoning was that potentially (although unlikely) we can span this test among more than one era in e2e tests, so we can be different in terms of fee comparison or total issuance different by `K*possibly_treasury_gain_from_staking`, where `K` is era span difference.

Reply:
I like this cautious approach, and I think it would be made better by making it more explicit. Could you create a function that takes all the needed data, asserts this, but then also logs the somewhat unexpected case when this is unequal, including how many eras have apparently passed (the `K`)? It would mostly make the code here much clearer, and sometimes it might end up being more useful by letting us know something unexpected happened.

Reply:
Sure will do.